homeassistant/components/sleepiq/button.py

```diff
@@ -0,0 +1,82 @@
+"""Support for SleepIQ buttons."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from asyncsleepiq import SleepIQBed
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SleepIQDataUpdateCoordinator
+from .entity import SleepIQEntity
+
+
+@dataclass
+class SleepIQButtonEntityDescriptionMixin:
+    """Describes a SleepIQ Button entity."""
+
+    press_action: Callable
+
+
+@dataclass
+class SleepIQButtonEntityDescription(
+    ButtonEntityDescription, SleepIQButtonEntityDescriptionMixin
+):
+    """Class to describe a Button entity."""
+
+
+ENTITY_DESCRIPTIONS = [
+    SleepIQButtonEntityDescription(
+        key="calibrate",
+        name="Calibrate",
+        press_action=lambda client: client.calibrate(),
+        icon="mdi:target",
+    ),
+    SleepIQButtonEntityDescription(
+        key="stoppump",
+        name="Stop Pump",
+        press_action=lambda client: client.stop_pump(),
+        icon="mdi:stop",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sleep number buttons."""
+    coordinator: SleepIQDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        SleepNumberButton(bed, ed)
+        for bed in coordinator.client.beds.values()
+        for ed in ENTITY_DESCRIPTIONS
+    )
+
+
+class SleepNumberButton(SleepIQEntity, ButtonEntity):
+    """Representation of an SleepIQ button."""
+
+    entity_description: SleepIQButtonEntityDescription
+
+    def __init__(
+        self, bed: SleepIQBed, entity_description: SleepIQButtonEntityDescription
+    ) -> None:
+        """Initialize the Button."""
+        super().__init__(bed)
+        self._attr_name = f"SleepNumber {bed.name} {entity_description.name}"
+        if entity_description.name:
```

Comment:
This shouldn't be needed anymore if you use .key

If mypy complains add an assert that description.key is not None